tests/basis_test.py

```diff
@@ -1,330 +1,1167 @@
-    f = ChebyshevFirstKind(4)
+
+import smolyay
+
+
+@pytest.mark.parametrize(
+    "cheb_function",
+    [
+        smolyay.basis.ChebyshevFirstKind,
+        smolyay.basis.ChebyshevSecondKind,
+    ],
+)
+def test_cheb_initial(cheb_function):
+    """test degrees returns correctly"""
+    f2 = cheb_function(2)
+    assert f2.degree == 2
+    assert isinstance(f2.degree, int)
+    assert numpy.array_equal(f2.domain, [-1, 1])
+    f2.degree = float(3)
+    assert f2.degree == 3
+    assert isinstance(f2.degree, int)
+
+
+@pytest.mark.parametrize(
+    "cheb_function,degree,answer_key",
+    [
+        (smolyay.basis.ChebyshevFirstKind, 0, {0.5: 1, 1: 1, -1: 1, -0.25: 1, -0.5: 1}),
+        (
+            smolyay.basis.ChebyshevFirstKind,
+            1,
+            {0.5: 0.5, 1: 1, -1: -1, -0.25: -0.25, -0.5: -0.5},
+        ),
+        (
+            smolyay.basis.ChebyshevFirstKind,
+            2,
+            {0.5: -0.5, 1: 1, -1: 1, -0.25: -0.875, -0.5: -0.5},
+        ),
+        (
+            smolyay.basis.ChebyshevSecondKind,
+            0,
+            {0.5: 1, 1: 1, -1: 1, -0.25: 1, -0.5: 1},
+        ),
+        (
+            smolyay.basis.ChebyshevSecondKind,
+            1,
+            {0.5: 1, 1: 2, -1: -2, -0.25: -0.5, -0.5: -1},
+        ),
+        (
+            smolyay.basis.ChebyshevSecondKind,
+            2,
+            {0.5: 0, 1: 3, -1: 3, -0.25: -0.75, -0.5: 0},
+        ),
+    ],
+)
+def test_cheb_call(cheb_function, degree, answer_key):
+    """Test chebyshev polynomial call"""
+    f = cheb_function(degree)
+    assert f(0.5) == pytest.approx(answer_key[0.5])
+    assert f(1) == pytest.approx(answer_key[1])
+    assert f(-1) == pytest.approx(answer_key[-1])
+    assert f(-0.25) == pytest.approx(answer_key[-0.25])
+    assert f(-0.5) == pytest.approx(answer_key[-0.5])
+
+
+@pytest.mark.parametrize(
+    "cheb_function,degree,answer_key",
+    [
+        (smolyay.basis.ChebyshevFirstKind, 0, {0.5: 1, 1: 1, -1: 1, -0.25: 1, -0.5: 1}),
+        (
+            smolyay.basis.ChebyshevFirstKind,
+            1,
+            {0.5: 0.5, 1: 1, -1: -1, -0.25: -0.25, -0.5: -0.5},
+        ),
+        (
+            smolyay.basis.ChebyshevFirstKind,
+            2,
+            {0.5: -0.5, 1: 1, -1: 1, -0.25: -0.875, -0.5: -0.5},
+        ),
+        (
+            smolyay.basis.ChebyshevSecondKind,
+            0,
+            {0.5: 1, 1: 1, -1: 1, -0.25: 1, -0.5: 1},
+        ),
+        (
+            smolyay.basis.ChebyshevSecondKind,
+            1,
+            {0.5: 1, 1: 2, -1: -2, -0.25: -0.5, -0.5: -1},
+        ),
+        (
+            smolyay.basis.ChebyshevSecondKind,
+            2,
+            {0.5: 0, 1: 3, -1: 3, -0.25: -0.75, -0.5: 0},
+        ),
+    ],
+)
+def test_cheb_call_1D(cheb_function, degree, answer_key):
+    """Test chebyshev polynomial call with a 1D array"""
+    f = cheb_function(degree)
+    xs = [0.5, 1, -1, -0.25, -0.5]
```

Comment:
This could be changed to:
```suggestion
    xs = list(answer_key.keys())
```
and this would also work with trig, right?